Resources/public/js/views/fields/ez-isbn-editview.js

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-isbn-editview', function (Y) {
+    "use strict";
+    /**
+     * Provides the field edit view for the ISBN field
+     *
+     * @module ez-isbn-editview
+     */
+    Y.namespace('eZ');
+
+    var FIELDTYPE_IDENTIFIER = 'ezisbn';
+
+    /**
+     * ISBN edit view
+     *
+     * @namespace eZ
+     * @class ISBNEditView
+     * @constructor
+     * @extends eZ.FieldEditView
+     */
+    Y.eZ.ISBNEditView = Y.Base.create('isbnEditView', Y.eZ.FieldEditView, [], {
+
+        events: {
+            '.ez-isbn-input-ui input': {
+                'blur': 'validate',
+                'valuechange': 'validate',
+            },
+        },
+
+        /**
+         * Validates the current input of the ISBN field
+         *
+         * @method validate
+         */
+        validate: function () {
+            this.set('errorStatus', false);
+            if ( this.get('fieldDefinition').isRequired && this._isFieldEmpty() ){
+                this.set('errorStatus', 'This field is required');
+            } else if ( !this._isFieldEmpty() ) {
+                if ( this.get('fieldDefinition').fieldSettings.isISBN13 ) {
+                    this._checkISBN13();
+                } else {
+                    this._checkISBN10();
+                }
+            }
+        },
+
+
+        /**
+         * Removes the slash of a given string
+         *
+         * @method _cleanDashes
+         * @param {String} string
+         * @protected
+         * @return {String}
+         */
+        _cleanDashes: function (string) {
```

Comment:
actually you write 3 times in the component something equivalent to `this._cleanDashes(this._getFieldValue())` and you call that *rawInputString*, it would then be better to introduce a method like `_getRawFieldValue` do that.